### PR TITLE
Add new `Style/MinMax` cop

### PR DESCRIPTION
**Pull request by Drenmi**
Just got back from vacation, so I wrote this small cop to get back into flow. It can be extended to also check for potential uses of `Enumerable#minmax_by`, but I don't have the time right now. 🙂 

This cop checks for potential uses of `Enumberable#minmax`, e.g.:

```
[foo.min, foo.max]
```

can be changed to:

```
foo.minmax
```

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests(`rake spec`) are passing.
* [X] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/min_max_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::MinMax, :config do
+  subject(:cop) { described_class.new(config) }
+
+  context 'with an array literal containing calls to `#min` and `#max`' do
+    it 'registers an offense if the receivers match' do
+      expect_offense(<<-RUBY.strip_indent)
```
Comment (bbatsov) [inline spec/rubocop/cop/style/min_max_spec.rb]: I guess you should add one example of `return foo.min, foo.max`.

**Comment by bbatsov** [comment]
I guess the inverse method doesn't exist right? I'm just wondering whether to flag `[foo.max, foo.min]` and suggest using `minmax` + `reverse`.

**Comment by Drenmi** [comment]
> I guess the inverse method doesn't exist right? I'm just wondering whether to flag [foo.max, foo.min] and suggest using minmax + reverse.

Yeah. It doesn't exist. I was actually thinking about this as well. Might add it. I think I'll do `#minmax_by` first.

File: spec/rubocop/cop/style/min_max_spec.rb (hunk 1 above)
Comment (Drenmi) [reply spec/rubocop/cop/style/min_max_spec.rb]: Ah. Yes. Perhaps parallel assignment, too. `bar = foo.min, foo.max`

File: spec/rubocop/cop/style/min_max_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        RUBY
+      end
+
+      it 'does not register an offense if the receivers do not match' do
+        expect_no_offenses(<<-RUBY.strip_indent)
+          [foo.min, bar.max]
+        RUBY
+      end
+
+      it 'does not register an offense if there are additional receivers' do
```
Comment (bquorning) [inline spec/rubocop/cop/style/min_max_spec.rb]: In this test there is only one receiver, `foo`. Perhaps reword as “does not register an offense if there are additional elements in the array”?

**Comment by bbatsov** [comment]
You should also take a look at those build failures.

File: spec/rubocop/cop/style/min_max_spec.rb (hunk 2 above)
Comment (Drenmi) [reply spec/rubocop/cop/style/min_max_spec.rb]: Good catch!

**Comment by Drenmi** [comment]
@bbatsov: Support for `return` added, build fixed, and branch rebased.

**Comment by bbatsov** [comment]
👍

**Pull request merged**